components/ai_chat/resources/page/components/main/index.tsx

```diff
@@ -57,8 +57,22 @@ function Main() {
     siteInfo === null && // SiteInfo request has finished and this is a standalone conversation
     !context.isPremiumUser
 
+  const isLastTurnBraveSearchSERPSummary = () => {
+    const lastTurn = context.conversationHistory.at(-1)
+
+    if (!lastTurn?.events?.length) {
+      return false
+    }
+
+    return lastTurn.events.some(
+      event => event.completionEvent?.fromBraveSearchSERP ?? false)
+  }
+
```

Comment:
Why is this a function in a function but not used as a callback? Why not just evaluate this case all the time, since it's immediately called just 2 lines later? Is it trying to be an optimization so that it doesn't run when there are no conversation entries? Doesn't seem expensive enough for that.

Reply:
Addressed in https://github.com/brave/brave-core/pull/25257/commits/138ebaf68057d3926f90b4bfeafa8cab2fd660b0.